src/cpp/src/visual_language/inputs_embedder.cpp

```diff
@@ -75,9 +89,11 @@ class InputsEmbedder::IInputsEmbedder {
     virtual void finish_chat() {
         m_is_chat_conversation = false;
         m_is_cache_empty = true;
+        m_trust_encoded_history = true;
 
         m_history.clear();
         m_templated_chat_history.clear();
+        m_tokenized_chat_history.clear();
```

Comment:
Keep the order of vars modification aligned between start and finish chat. Currently `start_chat()` clears `m_tokenized_chat_history` after `m_trust_encoded_history`, but `m_tokenized_chat_history` is cleared in the end in `finish_chat()`.